base/dict.jl

```diff
@@ -686,6 +686,115 @@ end
 
 filter!(f, d::Dict) = filter_in_one_pass!(f, d)
 
+
+
+
+
+
+
+"""
+    map!(f, values(dict))
+Takes the function f(value) and transforms values stored in the dict with the transformation value=f(value).
+
+
+# Examples
+```jldoctest
+julia> D=Dict(:a=>1,:b=>2)
+Dict{Symbol,Int64} with 2 entries:
+  :a => 1
+  :b => 2
+
+julia> map!(v->v-1,values(D))
+Dict{Symbol,Int64} with 2 entries:
+  :a => 0
+  :b => 1
+
+ ```
+"""
+Base.map!(f, iter::Base.ValueIterator{Dict{K,V}}) where {K, V}= (_map_to!(Val(false), f, iter.dict.vals, iter, iter.dict.idxfloor); iter.dict)
+
+
+"""map(f, values(dict), create_new_dict=true, value_type_change::Bool = true) -> dict
+Takes the function f(value) and creates a copy of the dict with the values transformed by value=f(value).
+Setting value_type_change=false will ensure that the values type of the input and output dicts are the same.
+create_new_dict=false will cause the map method to fallback to returning an array.
+
+# Examples
+```jldoctest
+julia> D=Dict(:a=>1, :b=>2, :c=>3)
+Dict{Symbol,Int64} with 3 entries:
+  :a => 1
+  :b => 2
+  :c => 3
+
+julia> E=map(v->isodd(v) ? Float64(v) : string(v), values(D), create_new_dict=true)
+Dict{Symbol,Union{Float64, String}} with 3 entries:
+  :a => 1.0
+  :b => "2"
+  :c => 3.0
+
+julia> E=map(v->isodd(v) ? Int64(v) : UInt32(v), values(D), create_new_dict=true)
+Dict{Symbol,Union{Int64, UInt32}} with 3 entries:
+  :a => 1
+  :b => 0x00000002
+  :c => 3
+
+julia> E=map(v->isodd(v) ? Int64(v) : UInt32(v), values(D), create_new_dict=true, value_type_change = false)
+Dict{Symbol,Int64} with 3 entries:
+  :a => 1
+  :b => 2
+  :c => 3
+ ```
+"""
+function Base.map(f,iter::Base.ValueIterator{Dict{K,V}}; create_new_dict::Bool, value_type_change::Bool=true) where {K, V}
```

Comment:
I would omit the `map` method, for which the API is still controversial, and only have `map!(f, values(dict))` in this PR